runtime/compiler/il/J9DataTypes.cpp

```diff
@@ -217,8 +215,8 @@ static_assert(TR::LastJ9Type - TR::FirstJ9Type + 1 == (sizeof(J9DataTypeSizes) /
 const int32_t
 J9::DataType::getSize(TR::DataType dt)
    {
-   TR_ASSERT(dt < TR::NumTypes, "dataTypeSizeMap called on unrecognized data type");
-   if (dt < TR::FirstJ9Type)
+   TR_ASSERT(dt < TR::NumAllTypes, "dataTypeSizeMap called on unrecognized data type");
+   if (dt < TR::FirstJ9Type || dt.isVector())
```

Comment:
I see this pattern repeated everywhere: would it be better to introduce a query like `isOMRDataType(TR::DataType dt)` and change all these to use that query rather than spraying `|| dt.isVector()` everywhere?

Reply:
Great idea. I will add  that query. Thanks!